fe/fe-core/src/main/java/org/apache/doris/job/manager/JobManager.java

```diff
@@ -0,0 +1,213 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.job.manager;
+
+import org.apache.doris.catalog.Env;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.common.util.LogBuilder;
+import org.apache.doris.common.util.LogKey;
+import org.apache.doris.job.base.AbstractJob;
+import org.apache.doris.job.common.JobStatus;
+import org.apache.doris.job.common.JobType;
+import org.apache.doris.job.exception.JobException;
+import org.apache.doris.job.scheduler.JobScheduler;
+import org.apache.doris.job.task.AbstractTask;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class JobManager<T extends AbstractJob<?>> implements Writable {
+
+
+    private final ConcurrentHashMap<Long, T> jobMap = new ConcurrentHashMap<>(32);
+
+    private JobScheduler jobScheduler;
+
+    public void start() {
+        jobScheduler = new JobScheduler(jobMap);
+        jobScheduler.start();
+    }
+
+    public Long registerJob(T job) throws JobException {
+        //job.checkJobParams();
+        checkJobNameExist(job.getJobName(), job.getJobType(), job.getCurrentDbName());
+        long id = Env.getCurrentEnv().getNextId();
```

Comment:
I suggest that the job id should be assigned when job is created.
Not in `registerJob()` method, so it will easier to write unit test